### PR TITLE
add NuGet.ProjectModel 6.2.2

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Part of https://github.com/dotnet/runtime/issues/81468

### CODE REVIEW

File: eng/Build.props [hunk 1]
```diff
@@ -80,6 +80,16 @@
 
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Build.Framework.15.1.1012.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Build.Utilities.Core.15.1.1012.csproj" />
+    
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\NuGet.Frameworks.6.2.2.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\NuGet.Common.6.2.2.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\NuGet.Versioning.6.2.2.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\NuGet.Packaging.6.2.2.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\NuGet.Configuration.6.2.2.csproj" />
```
[the remark this answers is not in the document]
Comment (oleksandr-didyk) [reply eng/Build.props:88]: Thanks, will fix

File: src/referencePackages/src/nuget.configuration/6.2.2/lib/net45/NuGet.Configuration.cs
[the remark this answers is not in the document]
Comment (oleksandr-didyk) [reply src/referencePackages/src/nuget.configuration/6.2.2/lib/net45/NuGet.Configuration.cs:1]: That's on me, I rebased on `main` and pulled these changes in after the packages were generated and manually removed the net45 references, but I missed the actual code. Thanks for pointing this out, will remove

File: src/referencePackages/src/nuget.common/6.2.2/nuget.common.nuspec [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <projectUrl>https://aka.ms/nugetprj</projectUrl>
+    <description>Common utilities and interfaces for all NuGet libraries.</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <tags>nuget</tags>
+    <serviceable>true</serviceable>
+    <repository type="git" url="https://github.com/NuGet/NuGet.Client" commit="027ca8b8ef4b4dc94995f87b9c441d2bcf742c1d" />
+    <dependencies>
+      <group targetFramework=".NETFramework4.5">
+        <dependency id="NuGet.Frameworks" version="6.2.2" exclude="Build,Analyzers" />
+      </group>
```
Comment (mthalman) [inline src/referencePackages/src/nuget.common/6.2.2/nuget.common.nuspec:19]: This can be removed.

**Review by mthalman — APPROVED** [review]

File: src/referencePackages/src/nuget.common/6.2.2/NuGet.Common.6.2.2.csproj [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net472;netstandard2.0</TargetFrameworks>
```
Comment (MichaelSimons) [inline src/referencePackages/src/nuget.common/6.2.2/NuGet.Common.6.2.2.csproj:4]: Can you regenerate these again off of the tip tooling.  I am not expecting and net4* tfms to be included by default anymore after a change I submitted yesterday.  If you simply regenerate, you will have to manually remove the net472 source files that were previously generated.

File: eng/Build.props [hunk 4]
```diff
@@ -44,7 +44,6 @@
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Reflection.MetadataLoadContext.6.0.0.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.Resources.Extensions.6.0.0.csproj" />
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.CodeDom.6.0.0.csproj" />
-    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.IO.Redist.6.0.0.csproj" />
```
Comment (oleksandr-didyk) [inline eng/Build.props:47]: This seems to be a left-over from one of the `4.*` tfm removals. It pops up in the pre-build report if left here

File: src/referencePackages/src/nuget.common/6.2.2/NuGet.Common.6.2.2.csproj (hunk 3 above)
Comment (MichaelSimons) [reply src/referencePackages/src/nuget.common/6.2.2/NuGet.Common.6.2.2.csproj]: I still see all of the net472 artifacts.  You will need to manually delete the net472 artifacts that were previously generated.

Comment (oleksandr-didyk) [reply src/referencePackages/src/nuget.common/6.2.2/NuGet.Common.6.2.2.csproj]: Yeah, I simply forgot to `git add` the removal. My bad, should be fixed now

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**